tests/current/contexts_precendence.yaml

```diff
@@ -0,0 +1,219 @@
+---
+tests:
+  # enabled tests
+  - cases:
+    - name: returns the correct `flag` value using the global context (1)
+      client: feature_flag_client
+      function: enabled
+      contexts:
+        global:
+          user:
+            isHuman: "verified"
+      input:
+        flag: "mixed.case.property.name"
+      expected:
+        value: true
+    - name: returns the correct `flag` value using the global context (2)
+      client: feature_flag_client
+      function: enabled
+      contexts:
+        global:
+          user:
+            isHuman: "?"
+      input:
+        flag: "mixed.case.property.name"
+      expected:
+        value: false
+    - name: returns the correct `flag` value when local context clobbers global context (1)
+      client: feature_flag_client
+      function: enabled
+      contexts:
+        global:
+          user:
+            isHuman: "?"
+        local:
+          user:
+            isHuman: "verified"
```

Comment:
@jkebinger this changes the format of how context is specified.

There are a couple ways to specify context, currently:

1. Put block context at the top of a test case (see `tests/current/enabled_with_contexts.yaml`)
2. Put context in the `input` node (e.g. tests/current/get_feature_flag.yaml)


When adding global context into the mix, the block context approach felt particularly annoying. This lead me to add a `contexts` node which can specify `global` and `block` (clearly, by name). Once that happened, it felt right to also pull the `input` `context` up into this `node` and name it `local`.

I'm curious what you think. I can convert the existing data to this format if it feels right.

FWIW, here's what the ruby changes look like to support this format (and keep backwards compat at the moment): https://github.com/prefab-cloud/prefab-cloud-ruby/commit/29f0da9c7c2536d8c9c511076a7a31c78adc482d

Reply:
That does feel right. 
I'd thought about adding global under `client_overrides:` just because that's already looked at for the client startup, which is where a global context would be needed, but it's no harder to check under contexts for a `global:`
To confirm, "block" is the thread-local-equivalent/implicit context?
If you're happy to standardize this, please do, and thank you

Reply:
Correct. `block` is exactly as you understand it. I'll standardize and make sure all the ruby tests pass. Thank you.